roles/setup/tasks/setup-container.yml

```diff
@@ -0,0 +1,84 @@
+---
+- name: Enable linger for {{ receptor_user }}
+  become: true
+  ansible.builtin.command:
+    cmd: "loginctl enable-linger {{ receptor_user }}"
+  changed_when: false
+
+- name: Get UID of {{ receptor_user }}
+  ansible.builtin.command:
+    cmd: "id -u {{ receptor_user }}"
+  register: receptor_uid
+  changed_when: false
+
+- name: Setup container
+  become: true
+  become_user: "{{ receptor_user }}"
+  environment:
+    XDG_RUNTIME_DIR: "/run/user/{{ receptor_uid.stdout }}"
+  vars:
+    receptor_uid_directory: "/run/user/{{ receptor_uid.stdout }}"
+  block:
+    - name: Enable podman socket
+      ansible.builtin.systemd:
+        name: podman.socket
+        state: started
+        enabled: true
+        scope: user
+
+    - name: Pull the latest release of receptor from Quay
+      containers.podman.podman_image:
+        name: "{{ receptor_image_name }}"
+        tag: "{{ receptor_image_tag }}"
+        state: present
+
+    - name: Run receptor image
+      containers.podman.podman_container:
+        name: "{{ receptor_service_name }}"
+        image: "{{ receptor_image_name }}:{{ receptor_image_tag }}"
+        state: created
+        network: host
+        pid: private
+        pids_limit: 0
+        cap_drop:
+          - CAP_AUDIT_WRITE
+          - CAP_MKNOD
+        user: root
+        volume:
+          - "{{ receptor_config_dir }}/receptor.conf:{{ receptor_config_dir }}/receptor.conf:ro"
+          - "{{ receptor_worksign_public_keyfile }}:{{ receptor_worksign_public_keyfile }}:ro"
+          - "{{ receptor_tls_certfile }}:{{ receptor_tls_certfile }}:ro"
+          - "{{ receptor_tls_keyfile }}:{{ receptor_tls_keyfile }}:ro"
+          - "{{ receptor_ca_certfile }}:{{ receptor_ca_certfile }}:ro"
+          - "{{ receptor_socket_dir }}:{{ receptor_socket_dir }}:z"
+          - "{{ receptor_uid_directory }}/podman/podman.sock:/run/podman/podman.sock:z"
+        privileged: true
+        security_opt:
+          - 'label=disable'
+        env:
+          RECEPTORCTL_SOCKET: '{{ receptor_socket_dir }}/{{ receptor_control_filename }}'
+        command: 'receptor --config {{ receptor_config_dir }}/receptor.conf'
+        generate_systemd:
+          container_prefix: ''
+          path: ~/.config/systemd/user/
+          separator: ''
+
+    - name: Ensure receptor service is started
+      ansible.builtin.systemd_service:
+        name: '{{ receptor_service_name }}'
+        state: started
+        enabled: true
+        scope: user
+        daemon_reload: true
+
+- name: Add 'receptorctl' alias
+  become: true
+  become_user: "{{ ansible_user }}"
+  ansible.builtin.lineinfile:
+    path: ~/.bashrc
```

Comment:
uh, and what if its using zsh ? 

So, if I understood correctly, this would add a bash alias to the `ansible_user` in case they are using bash to use sudo to move to awx and exec the command there.

What if `cloud-user` or whatever default user we use for `ansible_user` can't use sudo to impersonate `awx` ?

What if they won't to use `root` to call receptorctl ?

Won't be better an sh script or similar in /usr/local/bin or /usr/local/sbin that `root` or `awx` users could use to call receptorctl ?

Reply:
much better idea than an alias! 

@Zokormazo++

Reply:
Great note! Will update to capture this.
Will replace the alias with a script in `/usr/local/bin` which can be used to call receptorctl.

Reply:
I tried to address this again, but not entirely sure I understand the ask here after diving into this.

I added a script, which essentially just calls - `podman exec receptor receptor` to call the receptor, and stored. this in `/usr/local/bin/receptor`, and it is owned by awx user/group.
This essentially means it can only be called when logged in as the awx user.

Is that the intention here?